svgdigitizer/electrochemistry/electrolyte.py

```diff
@@ -0,0 +1,196 @@
+            1004
+
+            CID can be retrieved with InChI::
+            >>> Electrolyte.lookup_CID_from_InChI(InChI='InChI=1S/H3O4P/c1-5(2,3)4/h(H3,1,2,3,4)') # phosphoric acid
+            1004
+
+        """
+        if InChIKey:
+            return pcp.get_cids(InChIKey, 'inchikey')[0]
+        elif InChI:
+            return pcp.get_cids(InChI, 'inchi')[0]
+
+    @classmethod
+    def lookup_chemical_ids_from_name(cls, chemical_name):
+        r"""
+        Lookup CID, InChI, and InChIKey from chemical name.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> Electrolyte.lookup_chemical_ids_from_name('phosphoric acid')
+            {'CID': 1004, 'InChI': 'InChI=1S/H3O4P/c1-5(2,3)4/h(H3,1,2,3,4)', 'InChIKey': 'NBIIXXVUZAFLBC-UHFFFAOYSA-N'}
+
+        """
+        results = pcp.get_cids(chemical_name, 'name')
+        len_results = len(results)
+        if len_results > 1:
+            raise RuntimeError("Ambiguous chemical name! More than one chemical id return by pubchem. Use another chemical name")
+        elif len_results == 0:
+            raise RuntimeError(f"Name of chemical {chemical_name} not found in pubchem! Check spelling.")
+        else:
+            return pcp.get_properties(("InChIKey", "InChI"), results[0])[0]
+
+    def get_pKa(self, InChIKey):
+        r"""
+        Get pKa
+        Pubchem lookup only works for neutral acids!
+
+        """
+        pattern = re.compile(r'.*(?P<pKa>\d+\.\d*).*')
+        try:
+            return acid_base_dict[InChIKey]['components']
+        except KeyError:
+            cid = self.lookup_CID_from_InChI(InChIKey=InChIKey)
+            pKa_result = self.lookup_pKa(cid)['pKa']
+            # TODO support salts etc.
+            return [Acid(pKa=[pattern.match(i).group('pKa') for i in pKa_result.split(';')], charge=0)]
+
+    def lookup_pKa(self, cid):
+        r"""
+        get url from Pubchem to get pka lookup result
+        'XML' can be replaced with 'JSON' but it is harder to parse later on
+        for more info about Pubchem output types: https://pubchemdocs.ncbi.nlm.nih.gov/pug-rest$_Toc494865558
+        stolen from https://github.com/khoivan88/pka_lookup/blob/master/src/pka_lookup_pubchem.py
+        """
+        pka_lookup_result_xml = f'https://pubchem.ncbi.nlm.nih.gov/rest/pug_view/data/compound/{cid}/XML?heading=Dissociation+Constants'
+        headers = {
+            'user-agent': 'Mozilla/5.0 (X11; CentOS; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.75 Safari/537.36'}
+
+        # Get the html request info using CID number from pubchem
+        r = requests.get(pka_lookup_result_xml, headers=headers, timeout=15)
+        # Check to see if give OK status (200) and not redirect
+        if r.status_code == 200 and len(r.history) == 0:
```

Comment:
I think that this is not the way this should be done anymore in recent Python versions. (Maybe there is some reason why it should be done exactly like this?)

```suggestion
        from urllib.request import urlopen
        response = urlopen(f'https://pubchem.ncbi.nlm.nih.gov/rest/pug_view/data/compound/{cid}/XML?heading=Dissociation+Constants')
        tree = ET.fromstring(response.read())
```